py/desispec/ccdcalib.py

```diff
@@ -743,12 +799,50 @@ def make_dark_scripts(outdir, days=None, nights=None, cameras=None,
 
     #- Create exposure log so that N>>1 jobs don't step on each other
     nightlist = [int(tmp) for tmp in nights.split()]
+
+    if use_exptable:
+        #grab all exposures from the exposure log in case some have been marked bad
+        #note that some exposures will not be in here, so we'll assume those are all fine
+        log.info(f'Using exposure tables for {len(nightlist)} night directories')
+        expfiles=[]
+        for night in nightlist:
+            expfiles.append(get_exposure_table_pathname(night))
+        exptables = load_tables(expfiles)
+        exptable_all=table_vstack(exptables)
+        select = ((exptable_all['OBSTYPE']=='zero')|(exptable_all['OBSTYPE']=='dark'))
+        exptable_select=exptable_all[select]
+    
     log.info(f'Scanning {len(nightlist)} night directories')
     speclog = io.util.get_speclog(nightlist)
+    if use_exptable:
```

Comment:
As mentioned in another comment, I'd also add logic to make the `OBSTYPE`'s all one case (probably uppercase since that is what the "speclog" uses).

Reply:
done, please tell if there's a different preferred way on how to do this